examples/tools/file_reader.py

```diff
@@ -1,74 +1,133 @@
+import io
 from typing import Any, Literal
 
 from pydantic import ConfigDict
 
 from dynamiq.nodes import NodeGroup
+from dynamiq.nodes.agents.exceptions import ToolExecutionException
 from dynamiq.nodes.node import Node, ensure_config
 from dynamiq.runnables import RunnableConfig
 from dynamiq.utils.logger import logger
-from dynamiq.nodes.agents.exceptions import ToolExecutionException
 
+DESCRIPTION = """
+<tool_description>
+This tool reads content from multiple byte streams.
+Features:
+- Supports both bytes and BytesIO as input
+- Configurable to display partial content to manage output size
+- Handles multiple files in a single operation
+Input format:
+- List of files with byte content (bytes or BytesIO) with filename and description attributes.
+</tool_description>
+"""
 
-class FileReadTool(Node):
+
+class FileReaderTool(Node):
     """
-    A tool to read the content of a file.
+    A tool to read the content of one or more files from byte streams.
 
     Attributes:
         name (str): The name of the tool.
         description (str): The description of the tool.
-        file_path (str): The file path to read.
+        files (list[bytes | io.BytesIO] | None): List of files to read content from.
+        show_partial (bool): Whether to show only a part of the content to avoid overwhelming the output.
+        max_display_length (int): Maximum number of characters/bytes to display in the output.
     """
 
     group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
-    name: str = "Read a file's content"
-    description: str = "A tool that can be used to read a file's content from local storage."
-    file_path: str
+    name: str = "file-reader-tool"
+    description: str = DESCRIPTION
+    files: list[bytes | io.BytesIO] | None = None
+    show_partial: bool = True
```

Comment:
`show_partial` and `max_display_length` not clear for me in context of list of files. What a use-case of this fucntionality?

Reply:
Seems a bit redundant, will clean this